### PR TITLE
Feature/add diamond merkle erc1155

### PR DESCRIPTION
**Pull request by DinizSa**

### CODE REVIEW

File: packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansDiamond.sol [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansDiamond.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansFacet.sol [hunk 2]
```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansInit.sol [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansInit.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansInternal.sol [hunk 4]
```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansInternal.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansStorage.sol [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/arcadians/ArcadiansStorage.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/inventory/IInventoryFacet.sol [hunk 6]
```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: UNLICENSED
+
+/**
+ * Crated based in the following work:
+ * Authors: Moonstream DAO (engineering@moonstream.to)
+ * GitHub: https://github.com/G7DAO/contracts
+ */
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/inventory/IInventoryFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryFacet.sol [hunk 7]
```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: UNLICENSED
+
+/**
+ * Crated based in the following work:
+ * Authors: Moonstream DAO (engineering@moonstream.to)
+ * GitHub: https://github.com/G7DAO/contracts
+ */
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryInternal.sol [hunk 8]
```diff
@@ -0,0 +1,309 @@
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryInternal.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryStorage.sol [hunk 9]
```diff
@@ -0,0 +1,58 @@
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/inventory/InventoryStorage.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsDiamond.sol [hunk 10]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsDiamond.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol [hunk 11]
```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsInit.sol [hunk 12]
```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsInit.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsInternal.sol [hunk 13]
```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsInternal.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsStorage.sol [hunk 14]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsStorage.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleFacet.sol [hunk 15]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleInternal.sol [hunk 16]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleInternal.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleStorage.sol [hunk 17]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/merkle/MerkleStorage.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesFacet.sol [hunk 18]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesFacet.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesInternal.sol [hunk 19]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesInternal.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol [hunk 20]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol]: Always have a fixed solidity version

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+        public onlyMinter
+    {
+        _mint(to, id, amount);
+    }
+
+    function mintBatch(address to, uint256[] calldata ids, uint256[] calldata amounts)
+        public onlyMinter
+    {
+        _mintBatch(to, ids, amounts);
+    }
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: This function is open without any restriction/payment. I guess user needs to pay while minting a new item

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+
+    function claimBatch(uint256[] calldata tokenIds, uint[] calldata amounts, bytes32[][] calldata proofs) external nonReentrant {
+        _claimBatch(tokenIds, amounts, proofs);
+    }
+
+    function mint(address to, uint256 id, uint256 amount)
+        public onlyMinter
+    {
+        _mint(to, id, amount);
+    }
```
Comment (vr16x) [inline packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: This function is open without any restriction/payment. I guess user needs to pay while minting a new item

File: packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol (hunk 20 above)
Comment (DinizSa) [reply packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol]: @venkatesh16031999 this semantics `^versionNumber` only allows for minor versions updates, which are non-breaking.

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol (hunk 21 above)
Comment (DinizSa) [reply packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: It has the modifier `onlyMinter `, which is a role

File: packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol (hunk 22 above)
Comment (DinizSa) [reply packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: replied here: https://github.com/alto-io/opg-scaffold/pull/1#discussion_r1145595341

Comment (vr16x) [reply packages/solidity-ts/contracts/arcadiaDiamonds/items/ItemsFacet.sol]: Yes, sounds good. I have missed that

File: packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol (hunk 20 above)
Comment (vr16x) [reply packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol]: @DinizSa When new compiler versions are released, bugs can be introduced in the contract. We need to fix the contract version and there is no need for minor updates, right?

According to the SWC registry, it is a best practise to lock the pragma version 
ref: https://swcregistry.io/docs/SWC-103

Comment (DinizSa) [reply packages/solidity-ts/contracts/arcadiaDiamonds/roles/RolesStorage.sol]: Understood thanks, will lock the version

**Pull request closed without merging**